### PR TITLE
feat: Use 2024 vendor libraries that aren't deprecated

### PR DESCRIPTION
**Pull request by msoucy**

### CODE REVIEW

File: core/src/main/java/com/chopshop166/chopshoplib/sensors/gyro/SmartGyro.java [hunk 1]
```diff
@@ -1,10 +1,76 @@
...
 
 /** Gyro wrapper. */
-public interface SmartGyro extends Gyro {
+public interface SmartGyro {
+    /**
+     * Calibrate the gyro. It's important to make sure that the robot is not moving while the
+     * calibration is in progress, this is typically done when the robot is first turned on while
+     * it's sitting at rest before the match starts.
+     */
+    void calibrate();
```
Comment (bot190) [inline core/src/main/java/com/chopshop166/chopshoplib/sensors/gyro/SmartGyro.java:13]: We probably don't need this function..

File: core/src/main/java/com/chopshop166/chopshoplib/sensors/gyro/PigeonGyro2.java [hunk 2]
```diff
@@ -24,9 +25,7 @@ public PigeonGyro2(final Pigeon2 gyro) {
         this.gyro = gyro;
 
         // Automatically invert if the gyro is upside-down
-        final short accelerometerXYZ[] = new short[3];
-        this.gyro.getBiasedAccelerometer(accelerometerXYZ);
-        if (accelerometerXYZ[2] < 0) {
+        if (this.gyro.getRoll().getValueAsDouble() < 0) {
```
Comment (bot190) [inline core/src/main/java/com/chopshop166/chopshoplib/sensors/gyro/PigeonGyro2.java:28]: I think we can remove this and just use the on-device calibration to handle this.

File: core/src/main/java/com/chopshop166/chopshoplib/sensors/TalonFXEncoder.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+     * 
+     * @return The velocity in rpm.
+     */
+    @Override
+    public double getRate() {
+        return this.velocitySignal.getValueAsDouble();
+    }
+
+    @Override
+    public void reset() {
```
Comment (bot190) [inline core/src/main/java/com/chopshop166/chopshoplib/sensors/TalonFXEncoder.java:59]: 😦

**Review by bot190 — CHANGES_REQUESTED** [review]
I think we need to update the Pigeon2 code a bit, but otherwise this looks good.

File: core/src/main/java/com/chopshop166/chopshoplib/sensors/gyro/PigeonGyro2.java (hunk 2 above)
Comment (msoucy) [reply core/src/main/java/com/chopshop166/chopshoplib/sensors/gyro/PigeonGyro2.java:28]: Does this require a software change, or is it something we do in firmware?

Comment (bot190) [reply core/src/main/java/com/chopshop166/chopshoplib/sensors/gyro/PigeonGyro2.java:28]: The calibration is on the device, no software changes necessary. (Just remove all of this invert code)

**Review by bot190 — APPROVED** [review]

**Pull request merged**